flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -247,6 +247,14 @@ public ClusterInformation getClusterInformation() {
         }
     }
 
+    public List<CompletableFuture<ApplicationStatus>> getDispatcherShutdownFutures() {
+        synchronized (this) {
+            return dispatcherResourceManagerComponents.stream()
+                    .map(DispatcherResourceManagerComponent::getShutDownFuture)
+                    .collect(Collectors.toList());
+        }
+    }
```

Comment:
This looks a bit as if we are leaking implementation details for testing purposes. I am not sure whether this really good given that the `MiniCluster` is not a testing utility only. I'd rather go w/o if possible.

Reply:
removed